core/src/main/java/me/uport/sdk/core/NetworkConfiguration.kt

```diff
@@ -0,0 +1,159 @@
+package me.uport.sdk.core
+
+import org.komputing.khex.extensions.clean0xPrefix
+import org.komputing.khex.extensions.prepend0xPrefix
+import java.lang.IllegalStateException
+
+
+/**
+ * Network Configuration class which helps to hold URLs and addresses for different eth networks.
+ *
+ * When implementing this class you must supply an Infura Project ID
+ *
+ * use the `registerNetwork` method to override defaults
+ */
+object NetworkConfiguration {
```

Comment:
Why is a new singleton needed here?
It looks like `registerAllNetworksWithInfura(...)` would work just as well in the existing `Networks.kt`.
What am I missing here?